### PR TITLE
Vendir latest CRDs and fixes from testing with supervisor cluster

### PR DESCRIPTION
**Pull request by vijaykatam**
Signed-off-by: Vijay Katam <vkatam@vmware.com>

### What this PR does / why we need it

vendor latest CRDs and fix bugs + cleanup issues encountered while testing with supervisor cluster

### Describe testing done for PR

Tested with supervisor cluster
### Release note

```release-note
package-based-lcm: vendor latest CRDs and fix bugs + cleanup issues encountered while testing with supervisor cluster
```

### PR Checklist

- [x ] Squash the commits into one or a small number of logical commits
      
- [x ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: addons/controllers/addon_controller.go [hunk 1]
```diff
@@ -116,9 +116,9 @@ func (r *AddonReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ct
 		return ctrl.Result{}, err
 	}
 
-	tkrName, err := util.GetClusterLabel(cluster.Labels, constants.TKRLabel)
-	if err != nil {
-		return ctrl.Result{}, err
+	tkrName := util.GetClusterLabel(cluster.Labels, constants.TKRLabel)
```
Comment (vijaykatam) [inline addons/controllers/addon_controller.go:119]: This should not be an error and was flooding the logs.

File: addons/main.go [hunk 2]
```diff
@@ -159,7 +163,7 @@ func main() {
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
 		Scheme:                 scheme,
 		MetricsBindAddress:     flags.metricsAddr,
-		Port:                   9453,
+		Port:                   9865,
```
Comment (vijaykatam) [inline addons/main.go:166]: Supervisor cluster permits only certain ports(perhaps in 98xx range?). This one is available.

Comment (shyaamsn) [reply addons/main.go:166]: **thought**: Probably later we should make the port configurable.

**Review by shyaamsn — APPROVED** [review]
LGTM

**Review by danniel1205 — APPROVED** [review]

**Pull request merged**